components/pill/__examples__/listbox-icon.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { PropTypes } from 'react';
+import createReactClass from 'create-react-class';
+
+import Pill from '~/components/pill';
+import Icon from '~/components/icon';
+import IconSettings from '~/components/icon-settings';
+
+
+const PILLS = [{
+	category: 'standard',
+	name: 'account'
+
+}, {
+	category: 'standard',
+	name: 'case'
+}, {
+	category: 'utility',
+	name: 'retweet'
+}, {
+	category: 'standard',
+	name: 'solution'
+}, {
+	category: 'standard',
+	name: 'custom_notification'
+}, {
+	category: 'standard',
+	name: 'email'
+}, {
+	category: 'standard',
+	name: 'endorsement'
+}, {
+	category: 'standard',
+	name: 'recent'
+}, {
+	category: 'custom',
+	name: 'custom31'
+}];
+
+function noop () {}
+
+const Example = createReactClass({
+	displayName: 'PillWithIconListboxExample',
+
+	propTypes: {
+		action: PropTypes.func
+	},
+
+	getDefaultProps () {
+		return {
+			action: () => noop
+		};
+	},
+
+	getInitialState () {
+		return this.getAllOn();
+	},
+
+	render () {
+		return (
+			<IconSettings iconPath="/assets/icons">
+				<div>
+					<div className="slds-grid slds-grid_vertical-align-start">
+						<div className="slds-pill_container">
+							<ul
+								className="slds-listbox slds-listbox_horizontal slds-listbox_inline"
+								role="listbox"
+								aria-label="Selected Options:"
+								aria-orientation="horizontal"
+							>
+								{PILLS.map(this.renderListItem)}
+							</ul>
+						</div>
+					</div>
+				</div>
+			</IconSettings>
+		);
+	},
+
+	renderListItem (icon, index) {
+		if (this.state['pill' + index]) {
+			return (
+				<li
+					className="slds-listbox-item"
+					role="presentation"
+					key={index}
+				>
+					<Pill
+						labels={{
+							label: 'Pill Label',
+							title: 'Full pill label verbiage mirrored here',
+							removeTitle: 'Remove'
+						}}
+						assistiveText={{
+							remove: 'Press delete or backspace to remove'
+						}}
+						variant="option"
+						icon={
+							<Icon
+								title="Title"
+								category={icon.category}
+								name={icon.name}
+							/>
+						}
+						onClick={this.onClick}
+						onRemove={() => this.onRemove(event, 'pill' + index)}
+					/>
+
+				</li>
+			);
+		}
+		return null;
+	},
+
+	onClick (event) {
+		this.props.action('onClick')(event);
```

Comment:
There is a caveat in using the same interactive examples in both the story book and site stories (as I understand you suggested) in that `action` (`import { action } from '@storybook/addon-actions';`) is undefined in the site stories. The workaround I did was to have `action` exposed as a component property with a noop default. Perhaps there is a better way of doing it. Please let me know if there is a better course of action here.

Reply:
`noop` is fine. The more complicated option is [this](https://github.com/salesforce/design-system-react/blob/b60ddb3a4cf67103998a3552d5cdc804873978ac/components/combobox/__examples__/base-custom-menu-item.jsx#L56). Either way is fine.